modules/openapi-generator/src/main/resources/kotlin-client/libraries/multiplatform/api.mustache

```diff
@@ -6,28 +6,20 @@ package {{apiPackage}}
 
 import {{packageName}}.infrastructure.*
 import io.ktor.client.request.forms.formData
-import kotlinx.serialization.UnstableDefault
 import io.ktor.client.engine.HttpClientEngine
 import io.ktor.client.features.json.serializer.KotlinxSerializer
 import kotlinx.serialization.json.Json
-import kotlinx.serialization.json.JsonConfiguration
 import io.ktor.http.ParametersBuilder
 import kotlinx.serialization.*
-import kotlinx.serialization.internal.StringDescriptor
+import kotlinx.serialization.descriptors.*
+import kotlinx.serialization.encoding.*
 
 {{#operations}}
-{{#nonPublicApi}}internal {{/nonPublicApi}}class {{classname}} @UseExperimental(UnstableDefault::class) constructor(
-    baseUrl: kotlin.String = "{{{basePath}}}",
+{{#nonPublicApi}}internal {{/nonPublicApi}}class {{classname}}(
+    baseUrl: String = ApiClient.BASE_URL,
     httpClientEngine: HttpClientEngine? = null,
-    serializer: KotlinxSerializer
-) : ApiClient(baseUrl, httpClientEngine, serializer) {
-
-    @UseExperimental(UnstableDefault::class)
-    constructor(
-        baseUrl: kotlin.String = "{{{basePath}}}",
-        httpClientEngine: HttpClientEngine? = null,
-        jsonConfiguration: JsonConfiguration = JsonConfiguration.Default
-    ) : this(baseUrl, httpClientEngine, KotlinxSerializer(Json(jsonConfiguration)))
+    jsonSerializer: Json
```

Comment:
The creation of the KotlinxSerializer is now lazily done in the ApiClient::class.
The Json is used to pass down settings as they might vary depending on the api: 
```
Json {
            isLenient = true
            ignoreUnknownKeys = true
        }
```

Reply:
Now when instantiating some `Api` class, it's needed to pass `jsonSerializer`. It wasn't needed before this PR (`jsonConfiguration: JsonConfiguration = JsonConfiguration.Default` in the constructor that was removed), so I think that we should remain backward-compatible here and some kind of a sane default `Json` configuration would be good. Best if this default configuration is accessible for the client, so that they can customize it by building upon the default one.

I agree that e.g. `ignoreUnknownKeys = true` should be in this default. Otherwise if the service being called starts including some new properties, JSON deserialization will fail. No experience/opinion on `isLenient`.